sdk/core/core/inc/az_span.h

```diff
@@ -260,6 +260,34 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span span, uint64_t* out_number);
  */
 AZ_NODISCARD az_result az_span_to_uint32(az_span span, uint32_t* out_number);
 
+/**
+ * @brief az_span_find searches for `target` in `source`, returning an #az_span if it finds it.
+ *
+ * @param[in] source The #az_span containing the ASCII digits to be searched on.
+ * @param[in] target The #az_span containing the token to be searched in `source`.
+ * @param[out] out_span The #az_span pointing to the first occurrence of `target` in `source`, if it is found.
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if `target` is found in `source`
+ *          #AZ_ERROR_EOF if `target` is not found in `source`
+ * is found within the span.
+ */
+AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out_span);
+
+/**
+ * @brief az_span_token_next is a string tokenizer for az_span.
+ *
+ * @param[in] source The #az_span containing the ASCII digits to be searched on.
+ * @param[in] delimiter The #az_span containing the delimiter to "split" `source` into tokens.
+ * @param[out] out_token The #az_span pointing to the token delimited by the beginning of `source` 
+ *                       up to the first occurrence of `delimiter`, or the end of `source` 
```

Comment:
I think you should add that the delimiter itself is not included in out_token

Reply:
That's absolutely correct. I'll add that.